src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -282,25 +282,29 @@ Result AudioStreamAAudio::requestStop() {
 }
 
 // TODO: Update to return tuple of Result and framesWritten (avoids cast)
-int32_t AudioStreamAAudio::write(const void *buffer,
+ErrorOrValue<int32_t>   AudioStreamAAudio::write(const void *buffer,
                                      int32_t numFrames,
                                      int64_t timeoutNanoseconds) {
     AAudioStream *stream = mAAudioStream.load();
     if (stream != nullptr) {
-        return mLibLoader->stream_write(mAAudioStream, buffer, numFrames, timeoutNanoseconds);
+        int32_t result = mLibLoader->stream_write(mAAudioStream, buffer,
+                                                  numFrames, timeoutNanoseconds);
+        return ErrorOrValue<int32_t>(result);
```

Comment:
As I've noted, it's the responsibility of `AudioStreamAAudio::write` to implement the logic that considers whether the value of `result` indicates an error or not. `ErrorOrValue` must not know these details.

Reply:
Done